examples/python/cancellation/README.md

```diff
@@ -0,0 +1,183 @@
+### Cancelling RPCs
+
+RPCs may be cancelled by both the client and the server.
+
+#### The Example
+
+In the example, we implement a silly algorithm. We search for bytestrings whose
+hashes are similar to a given search string. For example, say we're looking for
+the string "doctor". Our algorithm may `JrqhZVkTDoctYrUlXDbL6pfYQHU=` or
+`RC9/7mlM3ldy4TdoctOc6WzYbO4=`. This is a brute force algorithm, so the server
+performing the search must be conscious the resources it allows to each client
+and each client must be conscientious of the resources demanded of the server.
+
+In particular, we ensure that client processes cancel the stream explicitly
+before terminating and we ensure the server cancels RPCs that have gone on longer
+than a certain number of iterations.
```

Comment:
What's the upside of adding the algorithm portion to the example? The usage of cancellation can be demonstrated easily through much easier examples, like helloworld, helloworld_stream.

I would suggest to keep this real world example, and implement something that has higher effective information ratio for our readers.

Reply:
Discussed offline. But I feel strongly that examples are meant to show not just *how* to use APIs, but also *when* to use APIs. The point of the algorithm is to teach potential users why they should use certain portions of our API.

I agree on the complexity though. I have moved the search algorithm to a separate file so that users won't have to look at it unless they're curious. People can simply treat is a black box `search` function with the properties described in the `README`.